File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:

        stack = []
        res = [0] * len(temperatures)

        for i, temp in enumerate(temperatures):

            while stack and temp > stack[-1][0]:
                val, index = stack.pop()
                
                res[index] = i - index
            
            stack.append((temp, i))
        
        return res



        

File: 0739-daily-temperatures/0739-daily-temperatures.cpp
#include <stack>;

class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int n = temperatures.size();
        std::stack<int> stack;
        std::vector<int> res(n, 0);

        for (int i = 0; i < n; i++)
        {

            while (!stack.empty() && temperatures[i] > temperatures[stack.top()])
            {
                int prevIndex = stack.top();
                stack.pop();
                res[prevIndex] = i - prevIndex;
            }
            stack.push(i);

        }
        return res;
    }
};